eclipse-collections-code-generator/src/main/resources/api/primitiveIterable.stg

```diff
@@ -79,17 +79,75 @@ public interface <name>Iterable extends PrimitiveIterable
     boolean contains(<type> value);
 
     /**
-     * Returns true if the all of the values specified in the source array are contained
+     * Returns true if all of the values specified in the source array are contained
      * in the <name>Iterable, and false if they are not.
      */
     boolean containsAll(<type>... source);
 
     /**
-     * Returns true if the all of the values specified in the source <name>Iterable are contained
+     * Returns true if all of the values specified in the source <name>Iterable are contained
      * in the <name>Iterable, and false if they are not.
      */
     boolean containsAll(<name>Iterable source);
 
+    /**
+     * Returns true if any of the values specified in the source array are contained
+     * in the <name>Iterable, and false if they are not.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(<type>... source)
+    {
+        for (<type> item : source)
+        {
+            if (this.contains(item))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if any of the values specified in the source <name>Iterable are contained
+     * in the <name>Iterable, and false if they are not.
+     *
+     * @since 11.0
+     */
+    default boolean containsAny(<name>Iterable source)
+    {
+        return source.anySatisfy(this::contains);
+    }
+
+    /**
+     * Returns true if none of the values specified in the source array are contained
+     * in the <name>Iterable, and false if they are.
+     *
+     * @since 11.0
+     */
+    default boolean containsNone(<type>... source)
+    {
+        for (<type> item : source)
```

Comment:
change this to an indexed for loop

Reply:
why? Arrays are the only types where the non-indexed for-loop has no overhead.

Reply:
I was thinking to keep it consistent in the entire codebase. Throughout we tend to prefer indexed for loops

Reply:
Changed to an indexed for loop.